### ISSUE TITLE
Incomplete listing of games

### ISSUE THREAD

**Issue by raynebc** [pathartl/BleemSync#116]
This is my first time using BleemSync, but perhaps I did something wrong.  I got a 64GB flash drive formatted as FAT32, named the flash drive SONY, created the required Games directory and created 20 sequentially numbered subfolders of Games (ie. "\Games\1", "\Games\2", ... "\Games\20").  I had some extra bin files to try later, so I created a "\Games\additionals" folder and put them there.  I used an online .cue file generator to make sure each of the twenty folders had a .cue file that referenced the .bin file with the correct name, with appropriate details for PS1 games:

`
FILE "[BIN FILENAME]" BINARY
  TRACK 01 MODE2/2352
    INDEX 01 00:00:00
`

I extracted the contents of the BleemSync-0.4.1-win7-x86.zip file onto the root of the flash drive, opened a command prompt to its BleemSync folder and ran BleemSync.exe.  It complained about my additionals folder by saying:

> Found game with invalid id: additionals

It didn't seem to do anything after this, so I used CTRL+C to stop it, moved my additionals folder to the root of the flash drive and re-ran BleemSync.exe.  It printed the following output:

`Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00754 from disc image file H:\BleemSync\..\Games\3\GameData\Bust-A-Move4 NTSC.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00067 from disc image file H:\BleemSync\..\Games\4\GameData\castlevania_-_symphony_of_the_night.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-01041 from disc image file H:\BleemSync\..\Games\5\GameData\chrono_cross_(disc_1).bin
Could not grab game info for serial SLUS-01041
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00543 from disc image file H:\BleemSync\..\Games\6\GameData\colony_wars_(disc_1).bin
Could not grab game info for serial SLUS-00543
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00745 from disc image file H:\BleemSync\..\Games\7\GameData\darkstalkers_3.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00922 from disc image file H:\BleemSync\..\Games\8\GameData\dino_crisis_(v1.1).bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00920 from disc image file H:\BleemSync\..\Games\9\GameData\Fear Effect (Disc 1).bin
Could not grab game info for serial SLUS-00920
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SCUS-94221 from disc image file H:\BleemSync\..\Games\10\GameData\final_fantasy_tactics.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SCUS-94457 from disc image file H:\BleemSync\..\Games\11\GameData\grandia_(disc_1).bin
Could not grab game info for serial SCUS-94457
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00708 from disc image file H:\BleemSync\..\Games\12\GameData\legacy_of_kain_-_soul_reaver.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00594 from disc image file H:\BleemSync\..\Games\13\GameData\metal_gear_solid_(v1.1)_(disc_1).bin
Could not grab game info for serial SLUS-00594
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00662 from disc image file H:\BleemSync\..\Games\14\GameData\parasite_eve_(disc_1).bin
Could not grab game info for serial SLUS-00662
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-01042 from disc image file H:\BleemSync\..\Games\15\GameData\parasite_eve_ii_(disc_1).bin
Could not grab game info for serial SLUS-01042
Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-01073 from disc image file H:\BleemSync\..\Games\16\GameData\rhapsody_-_a_musical_adventure.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00707 from disc image file H:\BleemSync\..\Games\17\GameData\silent_hill_(v1.1).bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00924 from disc image file H:\BleemSync\..\Games\18\GameData\star_trek_-_invasion.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00560 from disc image file H:\BleemSync\..\Games\19\GameData\tactics_ogre.bin

Game.ini doesn't exist, grabbing from BleemSync Central
Found serial SLUS-00405 from disc image file H:\BleemSync\..\Games\20\GameData\time_crisis.bin

Added game [1] Metal Gear Solid to the database
Added game [2] Tony Hawk's Pro Skater 2 to the database
Added game [-2147482647] Bust-a-Move 4 to the database
Added game [-2147482646] Castlevania Symphony of the Night to the database
Added game [-2147482645] Darkstalkers 3 to the database
Added game [-2147482644] Dino Crisis to the database
Added game [-2147482643] Final Fantasy Tactics to the database
Added game [-2147482642] Legacy of Kain - Soul Reaver to the database
Added game [-2147482641] Rhapsody - A Musical Adventure to the database
Added game [-2147482640] Silent Hill to the database
Added game [-2147482639] Star Trek - Invasion to the database
Added game [-2147482638] Tactics Ogre - Let Us Cling Together to the database
Added game [-2147482637] Time Crisis to the database
Successfully inserted 20 games
`

Maybe the first interrupted run of BleemSync caused problems, but the second run only identified 18 serial numbers, and even though it said it inserted 20 games, it only listed 13 games at the end of the execution.  Most of the game index numbers printed were overflowed signed numbers.

I connected the flash drive to the PS classic and connected one of the included controllers and powered on the console for the first time.  Various usage screens appeared and then the games were offered.  Of the 20 bin files on the flash drive, 13 games were displayed for play in the menu.  Two of them had broken thumbnails.  I launched a couple games and they generally seemed to work, but the face buttons weren't accepted by the emulated games even though they worked fine in the PS classic's main menu.

I power cycled the console and this time the buttons all seemed to work when I briefly tested a couple games.  I did find that Bust a Move 4 doesn't have its background music emulated correctly, but this is a problem with the emulator and not Bleem, because this was also a problem with Sony's release of the emulated game on the PSP.  I don't know if I should wipe the flash drive and give it another try, but if I can get all 20 games to appear it would be great.  Thanks in advance.

**Comment by raynebc** [comment]
Github isn't honoring the newlines in the .cue file code block above, it is three separate lines as expected.

**Comment by pathartl** [comment]
That's the autodetection not detecting some of the games. Please review [how to manually add games](https://github.com/pathartl/BleemSync#manual-game-configuration). Also make sure that your System/Defaults directory exists and has the pcsx.cfg from the ZIP file in it.

**Comment by raynebc** [comment]
I manually corrected the contents of the folders and re-ran BleemSync.  This time its only output was successfully inserting 20 games:

`Added game [1] Armored Core to the database
Added game [2] Blood Omen - Legacy of Kain to the database
Added game [3] Bust-a-Move 4 to the database
Added game [4] Castlevania Symphony of the Night to the database
Added game [5] Chrono Cross to the database
Added game [6] Colony Wars to the database
Added game [7] Darkstalkers 3 to the database
Added game [8] Dino Crisis to the database
Added game [9] Fear Effect to the database
Added game [10] Final Fantasy Tactics to the database
Added game [11] Grandia to the database
Added game [12] Legacy of Kain - Soul Reaver to the database
Added game [13] Metal Gear Solid to the database
Added game [14] Parasite Eve to the database
Added game [15] Parasite Eve 2 to the database
Added game [16] Rhapsody - A Musical Adventure to the database
Added game [17] Silent Hill to the database
Added game [18] Star Trek - Invasion to the database
Added game [19] Tactics Ogre - Let Us Cling Together to the database
Added game [20] Time Crisis to the database
Successfully inserted 20 games`

I cleanly unmounted the flash drive, plugged it into the PS classic and powered it on.  The PS1 loading screen and music played, but the power status light toggled between amber and green for a second or so then the console powered off.  From that point powering it on again would only display the 20 games that came with the PS classic instead of those on the flash drive.

I connected the flash drive to my computer and it complained about the unclean dismount from when the PS classic crashed.  I repaired the file system, verified all of the expected files were still present for the 20 games in the Games folder, then I deleted all of the folders except for \Games.  I extracted all of the folders from the BleemSync 0.4.1 release (excluding the Games folder) to the flash drive and ran BleemSync again.  It again reported the exact same output from earlier in this post where 20 games were inserted.  I ejected the flash drive from the computer again, plugged it into the PS classic and powered it on.  The same thing happened where the console powered off and plugging the flash drive into my computer again reports file system errors.

Did you have any advice for debugging this, or should I remove half of the games from the flash drive, re-run BleemSync and test until I've identified the individual game(s) that cause this to happen?  In case it matters, for the Bust a Move 4 game I mentioned having broken BGM, I replaced it with a version that has the separate bin file for each track (data track and audio track for each of the songs used in the game) and the suitable cue sheet.  Would BleemSync have a problem with a game that was one disc with multiple bin files?
